app/src/main/kotlin/no/nav/aap/oppgave/plukk/PlukkOppgaveService.kt

```diff
@@ -34,24 +36,35 @@ class PlukkOppgaveService(val connection: DBConnection, val enhetService: IEnhet
         val oppgaveRepo = OppgaveRepository(connection)
         val nesteOppgaver = oppgaveRepo.finnNesteOppgaver(filter.copy(enheter = enheter), maksAntallForsøk)
         for ((i, nesteOppgave) in nesteOppgaver.withIndex()) {
-            require(nesteOppgave.avklaringsbehovReferanse.referanse != null || nesteOppgave.avklaringsbehovReferanse.journalpostId != null) {
+            require(
+                nesteOppgave.avklaringsbehovReferanse.referanse != null ||
+                    nesteOppgave.avklaringsbehovReferanse.journalpostId != null
+            ) {
                 "AvklaringsbehovReferanse må ha referanse til enten behandling eller journalpost"
             }
 
+            val oppgaveId = OppgaveId(nesteOppgave.oppgaveId, nesteOppgave.oppgaveVersjon)
             val harTilgang = TilgangGateway.sjekkTilgang(nesteOppgave.avklaringsbehovReferanse, token)
             if (harTilgang) {
-                val oppgaveId = OppgaveId(nesteOppgave.oppgaveId, nesteOppgave.oppgaveVersjon)
                 oppgaveRepo.reserverOppgave(oppgaveId, ident, ident)
                 sendOppgaveStatusOppdatering(oppgaveId, HendelseType.RESERVERT, FlytJobbRepository(connection))
-                log.info("Fant neste oppgave med id ${nesteOppgave.oppgaveId} etter ${i + 1} forsøk for filterId $filterId")
+                log.info(
+                    "Fant neste oppgave med id ${nesteOppgave.oppgaveId} etter ${i + 1} forsøk for filterId $filterId"
+                )
                 return nesteOppgave
             }
+            avreserverHvisTilgangAvslått(oppgaveId = oppgaveId, ident = ident, oppgaveRepo = oppgaveRepo)
+            oppdaterUtdatertEnhet(oppgaveId = oppgaveId, oppgaveRepo = oppgaveRepo)
```

Comment:
Kodesnuttene i "plukkNesteOppgave" og "plukkOppgave" ser for meg like ut. Kan det trekkes ut til en felles funksjon som tar inn oppgaven, kanskje?